doc/inband-telemetry/SAI-Proposal-Advanced-Telemetry-draft.md

```diff
@@ -0,0 +1,740 @@
+{
+    ......
+    /**
+     * @brief INT endpoint
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_EP_ENABLE,
+
+    /**
+     * @brief INT transit
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_TRANSIT_ENABLE,
+
+    /**
+     * @brief Packet postcard
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_POSTCARD_ENABLE,
+
+    /**
+     * @brief Mirror on Drop
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_MIRROR_ON_DROP_ENABLE,
+
+    /**
+     * @brief Globally unique switch ID
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_SWITCH_ID,
+
+    /**
+     * @brief Telemetry report source IP address
+     *
+     * @type sai_ip4_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_SRC_IP,
+
+    /**
+     * @brief Telemetry report destination IP addresses
+     *
+     * @type sai_ip4_list_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_DST_IP_LIST,
+
+    /**
+     * @brief Telemetry report virtual router ID
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @default Switch default virtual router ID
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_VIRTUAL_ROUTER_ID,
+
+    /**
+     * @brief Telemetry report truncate size
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 256
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_TRUNCATE_SIZE,
+
+    /**
+     * @brief Telemetry report UDP destination port
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default: 9999
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_UDP_DST_PORT,
+
+    /**
+     * @brief Telemetry flow state clear cycle
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_FLOW_STATE_CLEAR_CYCLE,
+
+    /**
+     * @brief Latency sensitivity for flow state change detection
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 15
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_LATENCY_SENSITIVITY,
+
+    /**
+     * @brief INT sink downstream ports
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_SINK_PORT_LIST,
+
+    /**
+     * @brief Reserved DSCP value for INT over L4
+     *
+     * @type sai_ternary_field_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_L4_DSCP,
+
+} sai_switch_attr_t;
+~~~
+
+## Changes to `saiacl.h` for INT, Postcard and MoD watchlist
+~~~cpp
+typedef enum _sai_acl_action_type_t
+{
+    ......
+    /** Enable mirror on drop */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_MOD_ENABLE,
```

Comment:
do you have an example to enable MIRROR on drop? since it is an acl action, you also need a match entry, what is the match entry?

Reply:
Added an example